### PR TITLE
Benchmarking: Add pov_mode to V2 syntax

### PR DESCRIPTION
**Pull request by ggwpez**
Changes:
- Port the `pov_mode` attribute from the V1 syntax to V2
- Update `pallet-whitelist` and `frame-benchmarking-pallet-pov`

Follow up: also allow this attribute on top-level benchmark modules.

### CODE REVIEW

**Comment by ggwpez** [comment]
bot bench substrate-pallet --pallet=frame_benchmarking_pallet_pov

**Comment by ggwpez** [comment]
bot bench substrate-pallet --pallet=pallet_whitelist

File: substrate/frame/support/procedural/src/benchmark.rs [hunk 1]
```diff
@@ -72,51 +76,158 @@ struct RangeArgs {
...
+			PovModeAttr::parse(input).map(BenchmarkAttr::PoV)
+		} else {
+			Err(lookahead.error())
+		}
+	}
+}
+
+/// A `#[pov_mode = .. { .. }]` attribute.
+#[derive(Debug, Clone)]
+struct PovModeAttr {
```
Comment (pgherveou) [inline substrate/frame/support/procedural/src/benchmark.rs:135]: nit I would move the definition before the impl, I find it easier to scroll through the file this way

File: substrate/frame/support/procedural/src/benchmark.rs [hunk 2]
```diff
@@ -72,51 +76,158 @@ struct RangeArgs {
...
+		let _eq: Token![=] = input.parse()?;
+		let root = PovEstimationMode::parse(input)?;
+
+		let mut maybe_content = None;
+		let _ = || -> Result<()> {
+			let content;
+			syn::braced!(content in input);
+			maybe_content = Some(content);
+			Ok(())
+		}();
```
Comment (pgherveou) [inline substrate/frame/support/procedural/src/benchmark.rs:102]: why do you need to return a Result here?

**Review by pgherveou — APPROVED** [review]

File: substrate/frame/support/procedural/src/benchmark.rs (hunk 2 above)
Comment (ggwpez) [reply substrate/frame/support/procedural/src/benchmark.rs:102]: The `syn::braced!` macro just does a return 🙈  
Its internally using `parse_braces`, but that is private IIUC.

**Review by gupnik — APPROVED** [review]
Could we add some UI tests?

**Comment by ggwpez** [comment]
bot update-ui

**Pull request merged**